[bug] RetroArch 1.19.1 crashes on Wayland/Vulkan - "bogus present modes found"

### Logs

RetroArch log:
```
[INFO] [Config]: Looking for config in: "/home/user/.config/retroarch/retroarch.cfg".
[INFO] RetroArch 1.19.1 (Git 0792144fe3)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
[INFO] Capabilities: MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE42 AES AVX AVX2 
[INFO] Version: 1.19.1
[INFO] Git: 0792144fe3
[INFO] Built: Jun 16 2024
[INFO] =================================================
[INFO] [Input]: Found input driver: "x".
[ERROR] Couldn't find any cloud sync driver named ""
[INFO] Available cloud sync drivers are:
[INFO] 	webdav
[INFO] 	null
[WARN] Going to default to first cloud sync driver...
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Core]: Geometry: 320x240, Aspect: 1.333, FPS: 60.00, Sample rate: 48000.00 Hz.
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Set video size to: fullscreen.
[INFO] [Wayland]: fractional_scale_v1 enabled
[INFO] [Vulkan]: Vulkan dynamic library loaded.
[INFO] [Vulkan]: Found vulkan context: "vk_wayland".
[INFO] [Vulkan]: Detecting screen resolution: 2560x1600.
[INFO] [Vulkan]: Found GPU at index 0: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Using GPU index 0.
[INFO] [Vulkan]: Using semaphores for WSI acquire.
[INFO] [Vulkan]: Using GPU: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Queue family 0 supports 1 sub-queues.
[INFO] [Vulkan]: Got 3 swapchain images.
[INFO] [Wayland] Failed to specify monitor for fullscreen, letting compositor decide
[INFO] [Vulkan]: Using resolution 4480x2800.
[INFO] [Vulkan]: Using RGB565 format.
[INFO] [Vulkan]: Loading stock shader.
[INFO] [Joypad]: Found joypad driver: "udev".
[INFO] [Wayland]: Enabling idle inhibitor
[INFO] [Video]: Found display server: "null".
[INFO] [PulseAudio]: Requested 24576 bytes buffer, got 18432.
[INFO] [Audio]: Started synchronous audio driver.
[INFO] [ALSA] Using ALSA version 1.2.12
[INFO] [Microphone]: Initialized microphone driver.
[INFO] [Display]: Found display driver: "vulkan".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_music_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_video_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_image_history.lpl".
[INFO] [Playlist]: Loading favorites file: "/home/user/.config/retroarch/content_favorites.lpl".
[INFO] [PulseAudio]: Pausing.
[INFO] [Vulkan]: QueuePresent failed, destroying swapchain.
[ERROR] [Vulkan]: Bogus present modes found.
[ERROR] [Vulkan]: Bogus present modes found.
[ERROR] [Wayland/Vulkan]: Failed to update swapchain.
[INFO] [Config]: Saved new config to "/home/user/.config/retroarch/retroarch.cfg".
[INFO] [PulseAudio]: Unpausing.
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [PulseAudio]: Pausing.
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
```

### Bisect Results

I believe I'm hitting this case: https://github.com/libretro/RetroArch/blob/17c899fac3c18c92ea13a5ef7ccc168ff2987df7/gfx/common/vulkan_common.c#L2041

RetroArch 1.18.0 does not exhibit the same issue. Logs from that version (Opening and then manually closing it a few seconds later):
```
[INFO] [Config]: Looking for config in: "/home/user/.config/retroarch/retroarch.cfg".
[INFO] RetroArch 1.18.0 (Git 06fa5325f8)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
[INFO] Capabilities: MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE42 AES AVX AVX2 
[INFO] Version: 1.18.0
[INFO] Git: 06fa5325f8
[INFO] Built: May 22 2024
[INFO] =================================================
[INFO] [Input]: Found input driver: "x".
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Core]: Geometry: 320x240, Aspect: 1.333, FPS: 60.00, Sample rate: 48000.00 Hz.
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Set video size to: fullscreen.
[INFO] [Wayland]: fractional_scale_v1 enabled
[INFO] [Vulkan]: Vulkan dynamic library loaded.
[INFO] [Vulkan]: Found vulkan context: "vk_wayland".
[INFO] [Vulkan]: Detecting screen resolution: 2560x1600.
[INFO] [Vulkan]: Found GPU at index 0: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Using GPU index 0.
[INFO] [Vulkan]: Using semaphores for WSI acquire.
[INFO] [Vulkan]: Using GPU: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Queue family 0 supports 1 sub-queues.
[INFO] [Vulkan]: Got 3 swapchain images.
[INFO] [Wayland] Failed to specify monitor for fullscreen, letting compositor decide
[INFO] [Vulkan]: Using resolution 4480x2800.
[INFO] [Vulkan]: Using RGB565 format.
[INFO] [Vulkan]: Loading stock shader.
[INFO] [Joypad]: Found joypad driver: "udev".
[INFO] [Wayland]: Enabling idle inhibitor
[INFO] [Video]: Found display server: "null".
[INFO] [PulseAudio]: Requested 24576 bytes buffer, got 18432.
[INFO] [Audio]: Started synchronous audio driver.
[INFO] [ALSA] Using ALSA version 1.2.12
[INFO] [Microphone]: Initialized microphone driver.
[INFO] [Display]: Found display driver: "vulkan".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_music_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_video_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_image_history.lpl".
[INFO] [Playlist]: Loading favorites file: "/home/user/.config/retroarch/content_favorites.lpl".
[INFO] [Config]: Saved new config to "/home/user/.config/retroarch/retroarch.cfg".
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [PulseAudio]: Pausing.
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
```

### Environment information

- OS: Arch Linux
- Install method: pacman, official Arch Linux `[extra]` repo
- DE: COSMIC on Wayland

Here's the log from 1.18.0:
```
[INFO] [Config]: Looking for config in: "/home/user/.config/retroarch/retroarch.cfg".
[INFO] RetroArch 1.18.0 (Git 06fa5325f8)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
[INFO] Capabilities: MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE42 AES AVX AVX2 
[INFO] Version: 1.18.0
[INFO] Git: 06fa5325f8
[INFO] Built: May 22 2024
[INFO] =================================================
[INFO] [Input]: Found input driver: "x".
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Core]: Geometry: 320x240, Aspect: 1.333, FPS: 60.00, Sample rate: 48000.00 Hz.
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Set video size to: fullscreen.
[DEBUG] [Wayland]: Add global 1, interface wl_compositor, version 5
[DEBUG] [Wayland]: Add global 2, interface wl_subcompositor, version 1
[DEBUG] [Wayland]: Add global 3, interface wl_data_device_manager, version 3
[DEBUG] [Wayland]: Add global 4, interface ext_foreign_toplevel_list_v1, version 1
[DEBUG] [Wayland]: Add global 5, interface wp_fractional_scale_manager_v1, version 1
[DEBUG] [Wayland]: Add global 6, interface zwp_keyboard_shortcuts_inhibit_manager_v1, version 1
[DEBUG] [Wayland]: Add global 7, interface zxdg_output_manager_v1, version 3
[DEBUG] [Wayland]: Add global 8, interface zwlr_output_manager_v1, version 4
[DEBUG] [Wayland]: Add global 9, interface zcosmic_output_manager_v1, version 1
[DEBUG] [Wayland]: Add global 10, interface wp_presentation, version 1
[DEBUG] [Wayland]: Add global 11, interface zwp_primary_selection_device_manager_v1, version 1
[DEBUG] [Wayland]: Add global 12, interface zcosmic_output_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 13, interface zcosmic_workspace_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 14, interface zcosmic_toplevel_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 15, interface zcosmic_screencopy_manager_v2, version 1
[DEBUG] [Wayland]: Add global 16, interface wl_shm, version 1
[DEBUG] [Wayland]: Add global 17, interface wp_cursor_shape_manager_v1, version 1
[DEBUG] [Wayland]: Add global 18, interface wp_viewporter, version 1
[DEBUG] [Wayland]: Add global 19, interface org_kde_kwin_server_decoration_manager, version 1
[DEBUG] [Wayland]: Add global 20, interface zxdg_decoration_manager_v1, version 1
[DEBUG] [Wayland]: Add global 21, interface ext_session_lock_manager_v1, version 1
[DEBUG] [Wayland]: Add global 24, interface zwp_pointer_constraints_v1, version 1
[DEBUG] [Wayland]: Add global 25, interface zwp_pointer_gestures_v1, version 3
[DEBUG] [Wayland]: Add global 26, interface zwp_tablet_manager_v2, version 1
[DEBUG] [Wayland]: Add global 27, interface wp_security_context_manager_v1, version 1
[DEBUG] [Wayland]: Add global 28, interface zwp_input_method_manager_v2, version 1
[DEBUG] [Wayland]: Add global 29, interface zwp_text_input_manager_v3, version 1
[DEBUG] [Wayland]: Add global 30, interface zwp_virtual_keyboard_manager_v1, version 1
[DEBUG] [Wayland]: Add global 31, interface wp_alpha_modifier_v1, version 1
[DEBUG] [Wayland]: Add global 32, interface wp_single_pixel_buffer_manager_v1, version 1
[DEBUG] [Wayland]: Add global 33, interface ext_idle_notifier_v1, version 1
[DEBUG] [Wayland]: Add global 34, interface zwp_idle_inhibit_manager_v1, version 1
[DEBUG] [Wayland]: Add global 35, interface zwlr_layer_shell_v1, version 4
[DEBUG] [Wayland]: Add global 36, interface xdg_wm_base, version 6
[DEBUG] [Wayland]: Add global 37, interface xdg_activation_v1, version 1
[DEBUG] [Wayland]: Add global 38, interface zxdg_exporter_v2, version 1
[DEBUG] [Wayland]: Add global 39, interface zxdg_importer_v2, version 1
[DEBUG] [Wayland]: Add global 40, interface zcosmic_toplevel_info_v1, version 1
[DEBUG] [Wayland]: Add global 41, interface zcosmic_toplevel_manager_v1, version 2
[DEBUG] [Wayland]: Add global 42, interface zcosmic_workspace_manager_v1, version 2
[DEBUG] [Wayland]: Add global 43, interface zwp_relative_pointer_manager_v1, version 1
[DEBUG] [Wayland]: Add global 44, interface zwp_linux_dmabuf_v1, version 5
[DEBUG] [Wayland]: Add global 45, interface wl_drm, version 2
[DEBUG] [Wayland]: Add global 46, interface wp_drm_lease_device_v1, version 1
[DEBUG] [Wayland]: Add global 47, interface wl_output, version 4
[DEBUG] [Wayland]: Add global 48, interface wl_seat, version 9
[DEBUG] [Wayland]: Add global 51, interface wp_drm_lease_device_v1, version 1
[INFO] [Wayland]: fractional_scale_v1 enabled
[INFO] [Vulkan]: Vulkan dynamic library loaded.
[INFO] [Vulkan]: Found vulkan context: "vk_wayland".
[INFO] [Vulkan]: Detecting screen resolution: 2560x1600.
[INFO] [Vulkan]: Found GPU at index 0: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Using GPU index 0.
[INFO] [Vulkan]: Using semaphores for WSI acquire.
[INFO] [Vulkan]: Using GPU: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Queue family 0 supports 1 sub-queues.
[INFO] [Vulkan]: Got 3 swapchain images.
[INFO] [Wayland] Failed to specify monitor for fullscreen, letting compositor decide
[INFO] [Vulkan]: Using resolution 4480x2800.
[INFO] [Vulkan]: Using RGB565 format.
[INFO] [Vulkan]: Loading stock shader.
[INFO] [udev]: Pad #0 (/dev/input/event20) supports force feedback.
[INFO] [udev]: Pad #0 (/dev/input/event20) supports 16 force feedback effects.
[INFO] [Autoconf]: Xbox 360 Controller configured in port 1.
[INFO] [Joypad]: Found joypad driver: "udev".
[INFO] [Wayland]: Enabling idle inhibitor
[INFO] [Video]: Found display server: "null".
[DEBUG] [PulseAudio]: Sink detected: alsa_output.pci-0000_e6_00.3.Speakers
[DEBUG] [PulseAudio]: Sink detected: alsa_output.pci-0000_e6_00.3.Headphones
[DEBUG] [PulseAudio]: Sink detected: alsa_output.usb-Burr-Brown_from_TI_USB_Audio_CODEC-00.analog-stereo-output
[DEBUG] [PulseAudio]: Sink detected: alsa_output.usb-DisplayLink_Dell_Universal_Dock_D6000_2111205291-02.iec958-stereo
[INFO] [PulseAudio]: Requested 24576 bytes buffer, got 18432.
[INFO] [Audio]: Started synchronous audio driver.
[INFO] [ALSA] Using ALSA version 1.2.12
[INFO] [Microphone]: Initialized microphone driver.
[INFO] [Display]: Found display driver: "vulkan".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_music_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_video_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_image_history.lpl".
[INFO] [Playlist]: Loading favorites file: "/home/user/.config/retroarch/content_favorites.lpl".
[INFO] [Config]: Saved new config to "/home/user/.config/retroarch/retroarch.cfg".
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [PulseAudio]: Pausing.
[DEBUG] [Audio]: Stopped audio driver "pulse"
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
```

Here's the log from RetroArch 1.19.1:
```
[INFO] [Config]: Looking for config in: "/home/user/.config/retroarch/retroarch.cfg".
[INFO] RetroArch 1.19.1 (Git 0792144fe3)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
[INFO] Capabilities: MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE42 AES AVX AVX2 
[INFO] Version: 1.19.1
[INFO] Git: 0792144fe3
[INFO] Built: Jun 16 2024
[INFO] =================================================
[INFO] [Input]: Found input driver: "x".
[ERROR] Couldn't find any cloud sync driver named ""
[INFO] Available cloud sync drivers are:
[INFO] 	webdav
[INFO] 	null
[WARN] Going to default to first cloud sync driver...
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Core]: Geometry: 320x240, Aspect: 1.333, FPS: 60.00, Sample rate: 48000.00 Hz.
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Set video size to: fullscreen.
[DEBUG] [Wayland]: Add global 1, interface wl_compositor, version 5
[DEBUG] [Wayland]: Add global 2, interface wl_subcompositor, version 1
[DEBUG] [Wayland]: Add global 3, interface wl_data_device_manager, version 3
[DEBUG] [Wayland]: Add global 4, interface ext_foreign_toplevel_list_v1, version 1
[DEBUG] [Wayland]: Add global 5, interface wp_fractional_scale_manager_v1, version 1
[DEBUG] [Wayland]: Add global 6, interface zwp_keyboard_shortcuts_inhibit_manager_v1, version 1
[DEBUG] [Wayland]: Add global 7, interface zxdg_output_manager_v1, version 3
[DEBUG] [Wayland]: Add global 8, interface zwlr_output_manager_v1, version 4
[DEBUG] [Wayland]: Add global 9, interface zcosmic_output_manager_v1, version 1
[DEBUG] [Wayland]: Add global 10, interface wp_presentation, version 1
[DEBUG] [Wayland]: Add global 11, interface zwp_primary_selection_device_manager_v1, version 1
[DEBUG] [Wayland]: Add global 12, interface zcosmic_output_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 13, interface zcosmic_workspace_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 14, interface zcosmic_toplevel_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 15, interface zcosmic_screencopy_manager_v2, version 1
[DEBUG] [Wayland]: Add global 16, interface wl_shm, version 1
[DEBUG] [Wayland]: Add global 17, interface wp_cursor_shape_manager_v1, version 1
[DEBUG] [Wayland]: Add global 18, interface wp_viewporter, version 1
[DEBUG] [Wayland]: Add global 19, interface org_kde_kwin_server_decoration_manager, version 1
[DEBUG] [Wayland]: Add global 20, interface zxdg_decoration_manager_v1, version 1
[DEBUG] [Wayland]: Add global 21, interface ext_session_lock_manager_v1, version 1
[DEBUG] [Wayland]: Add global 24, interface zwp_pointer_constraints_v1, version 1
[DEBUG] [Wayland]: Add global 25, interface zwp_pointer_gestures_v1, version 3
[DEBUG] [Wayland]: Add global 26, interface zwp_tablet_manager_v2, version 1
[DEBUG] [Wayland]: Add global 27, interface wp_security_context_manager_v1, version 1
[DEBUG] [Wayland]: Add global 28, interface zwp_input_method_manager_v2, version 1
[DEBUG] [Wayland]: Add global 29, interface zwp_text_input_manager_v3, version 1
[DEBUG] [Wayland]: Add global 30, interface zwp_virtual_keyboard_manager_v1, version 1
[DEBUG] [Wayland]: Add global 31, interface wp_alpha_modifier_v1, version 1
[DEBUG] [Wayland]: Add global 32, interface wp_single_pixel_buffer_manager_v1, version 1
[DEBUG] [Wayland]: Add global 33, interface ext_idle_notifier_v1, version 1
[DEBUG] [Wayland]: Add global 34, interface zwp_idle_inhibit_manager_v1, version 1
[DEBUG] [Wayland]: Add global 35, interface zwlr_layer_shell_v1, version 4
[DEBUG] [Wayland]: Add global 36, interface xdg_wm_base, version 6
[DEBUG] [Wayland]: Add global 37, interface xdg_activation_v1, version 1
[DEBUG] [Wayland]: Add global 38, interface zxdg_exporter_v2, version 1
[DEBUG] [Wayland]: Add global 39, interface zxdg_importer_v2, version 1
[DEBUG] [Wayland]: Add global 40, interface zcosmic_toplevel_info_v1, version 1
[DEBUG] [Wayland]: Add global 41, interface zcosmic_toplevel_manager_v1, version 2
[DEBUG] [Wayland]: Add global 42, interface zcosmic_workspace_manager_v1, version 2
[DEBUG] [Wayland]: Add global 43, interface zwp_relative_pointer_manager_v1, version 1
[DEBUG] [Wayland]: Add global 44, interface zwp_linux_dmabuf_v1, version 5
[DEBUG] [Wayland]: Add global 45, interface wl_drm, version 2
[DEBUG] [Wayland]: Add global 46, interface wp_drm_lease_device_v1, version 1
[DEBUG] [Wayland]: Add global 47, interface wl_output, version 4
[DEBUG] [Wayland]: Add global 48, interface wl_seat, version 9
[DEBUG] [Wayland]: Add global 51, interface wp_drm_lease_device_v1, version 1
[INFO] [Wayland]: fractional_scale_v1 enabled
[INFO] [Vulkan]: Vulkan dynamic library loaded.
[INFO] [Vulkan]: Found vulkan context: "vk_wayland".
[INFO] [Vulkan]: Detecting screen resolution: 2560x1600.
[INFO] [Vulkan]: Found GPU at index 0: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Using GPU index 0.
[INFO] [Vulkan]: Using semaphores for WSI acquire.
[INFO] [Vulkan]: Using GPU: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Queue family 0 supports 1 sub-queues.
[INFO] [Vulkan]: Got 3 swapchain images.
[INFO] [Wayland] Failed to specify monitor for fullscreen, letting compositor decide
[INFO] [Vulkan]: Using resolution 4480x2800.
[INFO] [Vulkan]: Using RGB565 format.
[INFO] [Vulkan]: Loading stock shader.
[INFO] [udev]: Pad #0 (/dev/input/event20) supports force feedback.
[INFO] [udev]: Pad #0 (/dev/input/event20) supports 16 force feedback effects.
[INFO] [Autoconf]: Xbox 360 Controller configured in port 1.
[INFO] [Joypad]: Found joypad driver: "udev".
[INFO] [Wayland]: Enabling idle inhibitor
[INFO] [Video]: Found display server: "null".
[DEBUG] [PulseAudio]: Sink detected: alsa_output.pci-0000_e6_00.3.Speakers
[DEBUG] [PulseAudio]: Sink detected: alsa_output.pci-0000_e6_00.3.Headphones
[DEBUG] [PulseAudio]: Sink detected: alsa_output.usb-Burr-Brown_from_TI_USB_Audio_CODEC-00.analog-stereo-output
[DEBUG] [PulseAudio]: Sink detected: alsa_output.usb-DisplayLink_Dell_Universal_Dock_D6000_2111205291-02.iec958-stereo
[INFO] [PulseAudio]: Requested 24576 bytes buffer, got 18432.
[INFO] [Audio]: Started synchronous audio driver.
[INFO] [ALSA] Using ALSA version 1.2.12
[INFO] [Microphone]: Initialized microphone driver.
[INFO] [Display]: Found display driver: "vulkan".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_music_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_video_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_image_history.lpl".
[INFO] [Playlist]: Loading favorites file: "/home/user/.config/retroarch/content_favorites.lpl".
[DEBUG] [Audio]: Started audio driver "pulse" (is_shutdown=false)
[INFO] [PulseAudio]: Pausing.
[DEBUG] [Audio]: Stopped audio driver "pulse"
[INFO] [Vulkan]: QueuePresent failed, destroying swapchain.
[ERROR] [Vulkan]: Bogus present modes found.
[ERROR] [Vulkan]: Bogus present modes found.
[ERROR] [Wayland/Vulkan]: Failed to update swapchain.
[INFO] [Config]: Saved new config to "/home/user/.config/retroarch/retroarch.cfg".
[INFO] [PulseAudio]: Unpausing.
[DEBUG] [Audio]: Started audio driver "pulse" (is_shutdown=true)
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [PulseAudio]: Pausing.
[DEBUG] [Audio]: Stopped audio driver "pulse"
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[DEBUG] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
```

```
[INFO] [Config]: Looking for config in: "/home/user/.config/retroarch/retroarch.cfg".
[INFO] RetroArch 1.19.1 (Git 0c468f9a44)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
[INFO] Capabilities: MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE42 AES AVX AVX2 
[INFO] Version: 1.19.1
[INFO] Git: 0c468f9a44
[INFO] Built: Sep 21 2024
[INFO] =================================================
[INFO] [Input]: Found input driver: "x".
[ERROR] Couldn't find any cloud sync driver named ""
[INFO] Available cloud sync drivers are:
[INFO] 	webdav
[INFO] 	null
[WARN] Going to default to first cloud sync driver...
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Core]: Version of libretro API: 1, Compiled against API: 1
[INFO] [Core]: Geometry: 320x240, Aspect: 1.333, FPS: 60.00, Sample rate: 48000.00 Hz.
[INFO] [Video]: Set video size to: fullscreen.
[DEBUG] [Wayland]: Add global 1, interface wl_compositor, version 5
[DEBUG] [Wayland]: Add global 2, interface wl_subcompositor, version 1
[DEBUG] [Wayland]: Add global 3, interface wl_data_device_manager, version 3
[DEBUG] [Wayland]: Add global 4, interface ext_foreign_toplevel_list_v1, version 1
[DEBUG] [Wayland]: Add global 5, interface wp_fractional_scale_manager_v1, version 1
[DEBUG] [Wayland]: Add global 6, interface zwp_keyboard_shortcuts_inhibit_manager_v1, version 1
[DEBUG] [Wayland]: Add global 7, interface zxdg_output_manager_v1, version 3
[DEBUG] [Wayland]: Add global 8, interface zwlr_output_manager_v1, version 4
[DEBUG] [Wayland]: Add global 9, interface zcosmic_output_manager_v1, version 1
[DEBUG] [Wayland]: Add global 10, interface wp_presentation, version 1
[DEBUG] [Wayland]: Add global 11, interface zwp_primary_selection_device_manager_v1, version 1
[DEBUG] [Wayland]: Add global 12, interface zcosmic_output_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 13, interface zcosmic_workspace_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 14, interface zcosmic_toplevel_image_source_manager_v1, version 1
[DEBUG] [Wayland]: Add global 15, interface zcosmic_screencopy_manager_v2, version 1
[DEBUG] [Wayland]: Add global 16, interface wl_shm, version 1
[DEBUG] [Wayland]: Add global 17, interface wp_cursor_shape_manager_v1, version 1
[DEBUG] [Wayland]: Add global 18, interface wp_viewporter, version 1
[DEBUG] [Wayland]: Add global 19, interface org_kde_kwin_server_decoration_manager, version 1
[DEBUG] [Wayland]: Add global 20, interface zxdg_decoration_manager_v1, version 1
[DEBUG] [Wayland]: Add global 21, interface ext_session_lock_manager_v1, version 1
[DEBUG] [Wayland]: Add global 24, interface zwp_pointer_constraints_v1, version 1
[DEBUG] [Wayland]: Add global 25, interface zwp_pointer_gestures_v1, version 3
[DEBUG] [Wayland]: Add global 26, interface zwp_tablet_manager_v2, version 1
[DEBUG] [Wayland]: Add global 27, interface wp_security_context_manager_v1, version 1
[DEBUG] [Wayland]: Add global 28, interface zwp_input_method_manager_v2, version 1
[DEBUG] [Wayland]: Add global 29, interface zwp_text_input_manager_v3, version 1
[DEBUG] [Wayland]: Add global 30, interface zwp_virtual_keyboard_manager_v1, version 1
[DEBUG] [Wayland]: Add global 31, interface wp_alpha_modifier_v1, version 1
[DEBUG] [Wayland]: Add global 32, interface wp_single_pixel_buffer_manager_v1, version 1
[DEBUG] [Wayland]: Add global 33, interface ext_idle_notifier_v1, version 1
[DEBUG] [Wayland]: Add global 34, interface zwp_idle_inhibit_manager_v1, version 1
[DEBUG] [Wayland]: Add global 35, interface zwlr_layer_shell_v1, version 4
[DEBUG] [Wayland]: Add global 36, interface xdg_wm_base, version 6
[DEBUG] [Wayland]: Add global 37, interface xdg_activation_v1, version 1
[DEBUG] [Wayland]: Add global 38, interface zxdg_exporter_v2, version 1
[DEBUG] [Wayland]: Add global 39, interface zxdg_importer_v2, version 1
[DEBUG] [Wayland]: Add global 40, interface zcosmic_toplevel_info_v1, version 1
[DEBUG] [Wayland]: Add global 41, interface zcosmic_toplevel_manager_v1, version 2
[DEBUG] [Wayland]: Add global 42, interface zcosmic_workspace_manager_v1, version 2
[DEBUG] [Wayland]: Add global 43, interface zwp_relative_pointer_manager_v1, version 1
[DEBUG] [Wayland]: Add global 44, interface zwp_linux_dmabuf_v1, version 5
[DEBUG] [Wayland]: Add global 45, interface wl_drm, version 2
[DEBUG] [Wayland]: Add global 46, interface wp_drm_lease_device_v1, version 1
[DEBUG] [Wayland]: Add global 47, interface wl_output, version 4
[DEBUG] [Wayland]: Add global 48, interface wl_seat, version 9
[INFO] [Wayland]: fractional_scale_v1 enabled
[INFO] [Vulkan]: Vulkan dynamic library loaded.
[INFO] [Vulkan]: Found vulkan context: "vk_wayland".
[INFO] [Vulkan]: Detecting screen resolution: 720x400.
[INFO] [Vulkan]: Found GPU at index 0: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Using GPU index 0.
[INFO] [Vulkan]: Using semaphores for WSI acquire.
[INFO] [Vulkan]: Using GPU: "Intel(R) Iris(R) Plus Graphics (ICL GT2)".
[INFO] [Vulkan]: Queue family 0 supports 1 sub-queues.
[DEBUG] [Vulkan]: Swapchain supports present mode: MAILBOX.
[DEBUG] [Vulkan]: Swapchain supports present mode: FIFO.
[INFO] [Vulkan]: Got 3 swapchain images.
[INFO] [Wayland] Failed to specify monitor for fullscreen, letting compositor decide
[INFO] [Vulkan]: Using resolution 720x400.
[INFO] [Vulkan]: Using RGB565 format.
[INFO] [Vulkan]: Loading stock shader.
[DEBUG] [udev]: Couldn't open any joypads. Are permissions set correctly for /dev/input/event* and /run/udev/?
[INFO] [Joypad]: Found joypad driver: "udev".
[INFO] [Wayland]: Enabling idle inhibitor
[INFO] [Video]: Found display server: "null".
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[DEBUG] [PulseAudio]: Sink detected: alsa_output.pci-0000_e6_00.3.Speakers
[DEBUG] [PulseAudio]: Sink detected: alsa_output.pci-0000_e6_00.3.Headphones
[DEBUG] [PulseAudio]: Sink detected: alsa_output.pci-0000_00_1f.3.hdmi-stereo
[DEBUG] [PulseAudio]: Sink detected: alsa_output.usb-DisplayLink_Dell_Universal_Dock_D6000_2111205291-02.iec958-stereo
[DEBUG] [PulseAudio]: Sink detected: alsa_output.usb-Burr-Brown_from_TI_USB_Audio_CODEC-00.analog-stereo-output
[INFO] [PulseAudio]: Requested 24576 bytes buffer, got 18432.
[INFO] [Audio]: Started synchronous audio driver.
[INFO] [ALSA] Using ALSA version 1.2.12
[INFO] [Microphone]: Initialized microphone driver.
[INFO] [Display]: Found display driver: "vulkan".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_music_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_video_history.lpl".
[INFO] [Playlist]: Loading history file: "/home/user/.config/retroarch/content_image_history.lpl".
[INFO] [Playlist]: Loading favorites file: "/home/user/.config/retroarch/content_favorites.lpl".
[DEBUG] [Audio]: Started audio driver "pulse" (is_shutdown=false)
[INFO] [PulseAudio]: Pausing.
[DEBUG] [Audio]: Stopped audio driver "pulse"
[INFO] [Vulkan]: QueuePresent failed, destroying swapchain.
[ERROR] [Vulkan]: Bogus present modes found.
[ERROR] [Vulkan]: Bogus present modes found.
[ERROR] [Wayland/Vulkan]: Failed to update swapchain.
[INFO] [Config]: Saved new config to "/home/user/.config/retroarch/retroarch.cfg".
[INFO] [PulseAudio]: Unpausing.
[DEBUG] [Audio]: Started audio driver "pulse" (is_shutdown=true)
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [PulseAudio]: Pausing.
[DEBUG] [Audio]: Stopped audio driver "pulse"
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[DEBUG] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
```

--- a/gfx/common/wayland_common.c
+++ b/gfx/common/wayland_common.c
@@ -130,8 +130,12 @@ void xdg_toplevel_handle_configure_common(gfx_ctx_wayland_data_t *wl,
       wl->buffer_height = wl->fractional_scale ?
          FRACTIONAL_SCALE_MULT(wl->height, wl->fractional_scale_num) : wl->height * wl->buffer_scale;
       wl->resize        = true;
-      if (wl->viewport) /* Update viewport */
+      if (wl->viewport)
+      {
+         /* Stretch old buffer to fill new size, commit/roundtrip to apply */
          wp_viewport_set_destination(wl->viewport, wl->width, wl->height);
+         wl_surface_commit(wl->surface);
+      }
    }
 
    if (floating)
@@ -198,8 +202,12 @@ void libdecor_frame_handle_configure_common(struct libdecor_frame *frame,
       wl->buffer_height = wl->fractional_scale ?
          FRACTIONAL_SCALE_MULT(height, wl->fractional_scale_num) : height * wl->buffer_scale;
       wl->resize        = true;
-      if (wl->viewport) /* Update viewport */
+      if (wl->viewport)
+      {
+         /* Stretch old buffer to fill new size, commit/roundtrip to apply */
          wp_viewport_set_destination(wl->viewport, wl->width, wl->height);
+         wl_surface_commit(wl->surface);
+      }
    }
 
    state = wl->libdecor_state_new(wl->width, wl->height);
@@ -842,8 +850,12 @@ bool gfx_ctx_wl_set_video_mode_common_size(gfx_ctx_wayland_data_t *wl,
       wl->buffer_height        = wl->fractional_scale ?
          FRACTIONAL_SCALE_MULT(wl->buffer_height, wl->fractional_scale_num) : wl->buffer_height * wl->buffer_scale;
    }
-   if (wl->viewport) /* Update viewport */
+   if (wl->viewport)
+   {
+      /* Stretch old buffer to fill new size, commit/roundtrip to apply */
       wp_viewport_set_destination(wl->viewport, wl->width, wl->height);
+      wl_surface_commit(wl->surface);
+   }
 
 #ifdef HAVE_LIBDECOR_H
    if (wl->libdecor)

